compiler/rustc_lint/src/lints.rs

```diff
@@ -323,6 +323,13 @@ pub struct BuiltinTypeAliasGenericBounds<'a, 'b> {
     pub sub: Option<SuggestChangingAssocTypes<'a, 'b>>,
 }
 
+#[derive(LintDiagnostic)]
+#[diag(lint_macro_expr_fragment_specifier_2024_migration)]
+pub struct MacroExprFragment2023 {
```

Comment:
Presumably we want to name this...

```suggestion
pub struct MacroExprFragment2024 {
```

...with corresponding updates elsewhere.